### ISSUE TITLE
[Bug]: MO panic with runtime error: slice bounds out of range [:20] with capacity 0

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#4370]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):0.6.0
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

{"level":"ERROR","time":"2022/08/04 07:29:46.267667 +0800","caller":"logutil/api.go:63","msg":"tae driver: No space","stacktrace":"github.com/matrixorigin/matrixone/pkg/logutil.Errorf\n\t/home/sudong/mo-0.5.1/matrixone/pkg/li.go:63\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*appendableNode).ApplyAppend\n\t/home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/tables/node.go:308\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/tae/tlockAppender).ApplyAppend.func1.1\n\t/home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/tables/appender.go:106\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/tae/buffer.(*Node).Expand\n\t/home/sudong/mo-0.5.1/matrixone/pkg//tae/buffer/node.go:180\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*blockAppender).ApplyAppend.func1\n\t/home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/tables/appender.go:104\ngithub.com/matrixorigin/matr/vm/engine/tae/tables.(*appendableNode).DoWithPin\n\t/home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/tables/node.go:80\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*blockAppender).ApplyAppend\n\t/home/sudon1/matrixone/pkg/vm/engine/tae/tables/appender.go:101\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*localSegment).ApplyAppend\n\t/home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/txn/txnimpl/localseg.go:1b.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnTable).ApplyAppend\n\t/home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/txn/txnimpl/table.go:661\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpble).PreApplyCommit\n\t/home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/txn/txnimpl/table.go:683\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnDB).PreApplyCommit\n\t/home/sudong/mo-0.5.1/matrixone/pkne/tae/txn/txnimpl/txndb.go:362\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnStore).PreApplyCommit\n\t/home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/txn/txnimpl/store.go:391\ngithub.com/matrixorigone/pkg/vm/engine/tae/txn/txnbase.(*Txn).PreApplyCommit\n\t/home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/txn/txnbase/txn.go:201\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*TxnManager).onPreApplyComome/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/txn/txnbase/txnmgr.go:140\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*TxnManager).onPreparing\n\t/home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/txn/txmgr.go:183\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/sm.(*safeQueue).Start.func1\n\t/home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/logstore/sm/safeq.go:83"}

panic: runtime error: slice bounds out of range [:20] with capacity 0

goroutine 1122343585 [running]:
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/containers.(*vector[...]).ReadFromFile(0xc25826c690, {0x7f6a576c7d20, 0xc28d1431f0}, 0xc1e349b170)
 /home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/containers/vector.go:318 +0xad0
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*dataBlock).LoadColumnData(0xc2e6d600a0, 0x0, 0x2030b2?)
 /home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/tables/block.go:608 +0x106
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*dataBlock).ResolveColumnMVCCData(0xc2e6d600a0, 0x1914cb5, 0x0, 0x43de80?)
 /home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/tables/block.go:418 +0x5d
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*dataBlock).GetColumnDataById(0xc2e6d600a0, {0x14f8eb0?, 0xc30bed3d60?}, 0x2?, 0x7f6a878b5ab8?)
 /home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/tables/block.go:409 +0x92
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*dataBlock).GetColumnDataByName(0xc2e6d600a0, {0x14f8eb0, 0xc30bed3d60}, {0xc030e54178?, 0x10?}, 0x125cae0?)
 /home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/tables/block.go:399 +0x8a
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnBlock).GetColumnDataByName(0xc291098d50?, {0xc030e54178?, 0x30?}, 0xc2cacb8540?)
 /home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/txn/txnimpl/block.go:198 +0xbe
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/moengine.(*txnBlock).Read(0xc26452be28, {0x7f684ed6bfd0?, 0x0?, 0x7f6a878b8e48?}, {0xc2e01a3e40, 0x2, 0x2}, {0x0, 0x0, 0x0}, ...)
 /home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/moengine/block.go:38 +0x19a
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/moengine.(*txnReader).Read(0xc1ef744af0, {0xc2549a56e0, 0x2, 0x2}, {0xc2e01a3e40, 0x2, 0x2})
 /home/sudong/mo-0.5.1/matrixone/pkg/vm/engine/tae/moengine/reader.go:53 +0x1f3
github.com/matrixorigin/matrixone/pkg/vm/pipeline.(*Pipeline).Run(0xc26452bf20, {0x14be540, 0xc1ef744af0}, 0xc0395c8400)
 /home/sudong/mo-0.5.1/matrixone/pkg/vm/pipeline/pipeline.go:69 +0x1de
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).Run(0xc11e339ef0?, {0xc105691310?, 0xc042de9270?})
 /home/sudong/mo-0.5.1/matrixone/pkg/sql/compile/scope.go:318 +0xc5
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun.func1(0xc000f4f770?)
 /home/sudong/mo-0.5.1/matrixone/pkg/sql/compile/scope.go:336 +0x79
created by github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun
 /home/sudong/mo-0.5.1/matrixone/pkg/sql/compile/scope.go:331 +0x1ad

**Comment by sukki37** [comment]
@LeftHandCold any status update?

**Comment by LeftHandCold** [comment]
fixed

**Comment by aressu1985** [comment]
fixed: b9c225f4f44dcd62d6f577394d705b6aed55188e
